csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1061 @@
+            } else if (is(returnType == wstring)) {
+                funcStr ~= "            return SharedFunctions.SliceToString(dlang_ret, DStringType._wstring);" ~ newline;
+            } else if (is(returnType == dstring)) {
+                funcStr ~= "            return SharedFunctions.SliceToString(dlang_ret, DStringType._dstring);" ~ newline;
+            } else if (isArray!(returnType)) {
+                funcStr ~= "            return new Range<%s>(dlang_ret);".format(getCSharpInterfaceType(returnTypeStr)) ~ newline;
+            } else {
+                funcStr ~= "            return dlang_ret;" ~ newline;
+            }
+        }
+        funcStr ~= "        }" ~ newline;
+        ns.functions ~= funcStr;
+    }
+
+    char[] ret;
+    ret.reserve(33_554_432);
+    foreach(csns; namespaces.byValue()) {
+        ret ~= csns.toString();
+    }
+
+    ret ~= newline ~ writeCSharpBoilerplate(libraryName, rootNamespace);
+
+    return cast(immutable)ret;
+}
+
+private string getDLangInterfaceType(string type) {
+    //Types that require special marshalling types
+    if (type[$-2..$] == "[]") return "slice";
+    else if (type == stringTypeString) return "slice";
+    else if (type == wstringTypeString) return "slice";
+    else if (type == dstringTypeString) return "slice";
+    else if (type == boolTypeString) return "bool";
+    //else if (type == dateTimeTypeString) return "datetime";
+    //else if (type == sysTimeTypeString) return "systime";
+    //else if (type == uuidTypeString) return "uuid";
+
+    //Types that can be marshalled by default
+    else if (type == charTypeString) return "byte";
+    else if (type == wcharTypeString) return "char";
+    else if (type == dcharTypeString) return "uint";
+    else if (type == ubyteTypeString) return "byte";
+    else if (type == byteTypeString) return "sbyte";
+    else if (type == ushortTypeString) return "ushort";
+    else if (type == shortTypeString) return "short";
+    else if (type == uintTypeString) return "uint";
+    else if (type == intTypeString) return "int";
+    else if (type == ulongTypeString) return "ulong";
+    else if (type == longTypeString) return "long";
+    else if (type == floatTypeString) return "float";
+    else if (type == doubleTypeString) return "double";
+    else return getCSharpName(type);
+}
+
+private string getCSharpInterfaceType(string type) {
+    //Types that require special marshalling types
+    if (type[$-2..$] == "[]") type = type[0..$-2];
+    if(type == voidTypeString) return "void";
+    else if(type == stringTypeString) return "string";
+    else if (type == wstringTypeString) return "string";
+    else if (type == dstringTypeString) return "string";
+    else if (type == boolTypeString) return "bool";
+    //else if (type == dateTimeTypeString) return "DateTime";
+    //else if (type == sysTimeTypeString) return "DateTimeOffset";
+    //else if (type == uuidTypeString) return "Guid" 
```

Comment:
?

Unexplained commented out code is bad (god knows I've been guilty...)

Reply:
Yea, that was stuff I was going to get to and then I got redirected and forgot it was there.